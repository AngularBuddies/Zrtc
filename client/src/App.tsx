import axios from 'axios';
import Peer from 'simple-peer';
import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {


  const sendOffer = () => {
    const peer = new Peer({
      initiator: true,
      trickle: false,
    });

    peer.on('error', console.error.bind(console, 'error'));
    peer.on('signal', (offer) => {
      console.log('signal', offer);
      axios.post('/offer', offer)
        .then(console.log.bind(console, 'offer success'))
        .catch(console.error.bind(console, 'offer error'))
    });
    peer.on('connect', console.log.bind(console, 'connect'));
    peer.on('data', console.log.bind(console, 'data'));
  }

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn RTS
        </a>
        <button onClick={sendOffer} >SendOffer</button>
      </header>
    </div>
  );
}

export default App;
