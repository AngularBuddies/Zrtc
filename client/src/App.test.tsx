import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('App', () => {
  beforeEach(() => {
    global.navigator.getUserMedia = jest.fn();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
