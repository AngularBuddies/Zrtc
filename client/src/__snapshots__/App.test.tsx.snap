// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        Stir Cray? Video Call with Half Decent People
      </h1>
      <a
        href="https://github.com/AngularBuddies/Zrtc"
      >
        Check out the source on Github.
      </a>
      <div
        class="buttons"
      >
        <button>
          SendOffer
        </button>
        <button>
          GetOffer
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
